### PR TITLE
TINY-4780: Enable spacing lints

### PR DESCRIPTION
**Pull request by tiny-james**
Enabled lints:
@typescript-eslint/indent
array-bracket-spacing
object-curly-spacing
space-before-function-paren
spaced-comment

### CODE REVIEW

File: modules/tinymce/src/themes/mobile/main/ts/ui/SerialisedDialog.ts [hunk 1]
```diff
@@ -120,14 +120,14 @@ const sketch = function (rawSpec) {
           }),
           Keying.config({
             mode: 'special',
-            focusIn (dialog/*, specialInfo */) {
+            focusIn(dialog/* , specialInfo */) {
```
Comment (tiny-james) [inline modules/tinymce/src/themes/mobile/main/ts/ui/SerialisedDialog.ts]: This comment spacing is a bit weird - I could change this to `focusIn(dialog, _specialInfo) {` to get it to ignore the unused variable.

File: modules/tinymce/src/themes/mobile/main/ts/util/Thor.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { Attr, Css, SelectorFilter } from '@ephox/sugar';
 const attr = 'data-ephox-mobile-fullscreen-style';
 const siblingStyles = 'display:none!important;';
 const ancestorPosition = 'position:absolute!important;';
-/// TINY-3407 ancestors need 'height:100%!important;overflow:visible!important;' to prevent collapsed ancestors hiding the editor
+// / TINY-3407 ancestors need 'height:100%!important;overflow:visible!important;' to prevent collapsed ancestors hiding the editor
```
Comment (tiny-james) [inline modules/tinymce/src/themes/mobile/main/ts/util/Thor.ts]: Does `///` have special meaning? If so then this could be a problem.

Comment (EkimChau) [reply modules/tinymce/src/themes/mobile/main/ts/util/Thor.ts]: not that i know of, appears to be a left over typo

Comment (tiny-james) [reply modules/tinymce/src/themes/mobile/main/ts/util/Thor.ts]: There were a few scattered through the code. I've removed the extra slash.

File: modules/tinymce/src/themes/mobile/main/ts/ui/SerialisedDialog.ts (hunk 1 above)
Comment (tiny-james) [reply modules/tinymce/src/themes/mobile/main/ts/ui/SerialisedDialog.ts]: I converted all commented out parameters into either unused parameters (using leading underscore) or I removed them.

**Pull request merged**